### PR TITLE
Capture remaining binding Bind & Get methods for source generator support

### PR DESCRIPTION
**Pull request by layomia**
Contributes to https://github.com/dotnet/runtime/issues/83600. Captures all `Bind` and `Get` overloads on `ConfigurationBinder` itself.

Follow-up PRs:

- [x] `ConfigurationBinder.GetValue` methods (https://github.com/dotnet/runtime/pull/86076)
- [ ] `M.Extensions.Options`-provided methods

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.cs [hunk 1]
```diff
@@ -42,29 +49,30 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
...
+                {
+                    Emitter emitter = new(context, spec);
+                    emitter.Emit();
+                }
+            }
+            catch (System.Exception ex)
             {
-                Emitter emitter = new(context, spec);
-                emitter.Emit();
+                ex.ToString();
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.cs:75]: >  ex.ToString(); [](http://example.com/codeflow?start=14&length=16)

How this is helping in anything?

File: src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs [hunk 2]
```diff
@@ -58,27 +126,54 @@ public Emitter(SourceProductionContext context, SourceGenerationSpec generationS
 
             public void Emit()
             {
+                if (!_generationSpec.ShouldEmitMethods(Helpers.GetMethods | Helpers.BindMethods | MethodSpecifier.Configure))
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs:129]: >Configure [](http://example.com/codeflow?start=114&length=9)

we don't need GetValue here?

File: src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.cs (hunk 1 above)
Comment (layomia) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.cs:75]: Was for debugging locally. Will remove/clean up.

File: src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs [hunk 3]
```diff
@@ -58,27 +126,54 @@ public Emitter(SourceProductionContext context, SourceGenerationSpec generationS
...
+                EmitGetValueMethods();
                 EmitBindMethods();
+
                 _writer.WriteBlockEnd();
 
+                _writer.WriteBlankLine();
+                #endregion
+
+                #region Helper class in source-generation namespace.
+                _useFullyQualifiedNames = false;
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs:153]: >_useFullyQualifiedNames [](http://example.com/codeflow?start=16&length=23)

How we decide to turn on or off this?

**Review by tarekgh — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs (hunk 2 above)
Comment (layomia) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs:129]: Implementation for `GetValue`'s coming in a follow up. For now, we throw `NotSupportedException`.

**Pull request merged**